### ISSUE TITLE
Double in value type is not allowed

### ISSUE THREAD

**Issue by sserbin** [vimeo/psalm#1099]
Test case:
```php
<?php
  
/**
 * @param 1|'1'|1.0 $a
 */
function takesOne($a): bool
{
    return $a === 1;
}

takesOne(1.0);
```

Expected: no errors
Actual: ERROR: InvalidDocblock - 4:11 - Unexpected token . in docblock for takesOne

https://getpsalm.org/r/2a3613d033
